e2e/cypress/integration/team_settings/teams_spec.js

```diff
@@ -211,4 +220,175 @@ describe('Teams Suite', () => {
         // # Close the team settings
         cy.get('body').type('{esc}', {force: true});
     });
+
+    it('MM-T2312 Team setting / Team name: Change name', () => {
+        // # visit /
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click();
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team name menu item
+        cy.findByText('Team Name').should('be.visible').click();
+
+        // # Change team name in the input
+        cy.get('#teamName').should('be.visible').clear().type('Test team');
+
+        // Save new team name
+        cy.get('#saveSetting').click();
```

Comment:
Thanks for the recommendation. Isn't finding by id the best option given the uniqueness of the id's? Seems to me that since this is a largely used method, we might need a separate ticket to change the implementation and also get rid of the ID's on the HTML elements if that's the case.

Reply:
Hello Nevy, yes they're are unique but it leads to test approach which is more on implementation details. For some time already, we're following the philosophy of [testing-library](https://testing-library.com/) where it test the app the way the software is used. There are still some use case for IDs but we're shifting gradually or as much as possible. I can create a ticket about migration the test cases but it'll be a low priority currently.